### PR TITLE
SAML metadata test cases

### PR DESCRIPTION
**Pull request by PasinduTennage**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+    private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp, String metadataXML) {
+
+        FederatedAuthenticatorConfig saml2SSOAuthnConfig = new FederatedAuthenticatorConfig();
+        saml2SSOAuthnConfig.setName("SAMLSSOAuthenticator");
+        saml2SSOAuthnConfig.setDisplayName("samlsso");
+        saml2SSOAuthnConfig.setEnabled(true);
+        Property[] properties = new Property[25];
+
+        Property property = new Property();
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java]: why do we need all these configs? If we upload a metadata file aren't we configuring the "Service Provider Entity Id" only?

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 2]
```diff
@@ -111,15 +110,22 @@
             <class name="org.wso2.identity.integration.test.user.profile.mgt.UserProfileAdminTestCase"/>
             <!--SCIM Test Cases-->
             <!--<class name="org.wso2.identity.integration.test.scim.SCIMUserProviderTestCase" />-->
+
+            <class name="org.wso2.identity.integration.test.scim.SCIMServiceProviderUserTestCase"/>
+            <class name="org.wso2.identity.integration.test.scim.SCIMServiceProviderGroupTestCase"/>
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Remove these 2. They are already defined below.

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 3]
```diff
@@ -128,8 +134,13 @@
 
             <!--Identity Provider Management Test Cases-->
             <class name="org.wso2.identity.integration.test.idp.mgt.IdentityProviderManagementTestCase"/>
+
+            <class name="org.wso2.identity.integration.test.idp.mgt.IdentityProviderMgtServiceTestCase"/>
+            <class name="org.wso2.identity.integration.test.requestPathAuthenticator.RequestPathAuthenticatorTestCase"/>
+
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Remove these 2. They are already defined below.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+                break;
+            }
+        }
+        Assert.assertNotNull(identityProvider);
+        Assert.assertNotNull(samlFederatedAuthenticatorConfig);
+
+        Property [] properties = samlFederatedAuthenticatorConfig.getProperties();
+        for (Property property : properties){
+
+            if(property.getName().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.IDP_ENTITY_ID)){
```
Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java]: Fix formatting issues throughout the PR

Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java]: Fix formatting issues throughout the PR

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java (hunk 1 above)
Comment (darshanasbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/IDPMetadataTestCase.java]: It seems like, it set some default values and check whether those defaults get overridden by the configs given in the XML.

More cleaner way is to do this in two steps. One create idp with some defaults, and then upload metadata file and check defaults get changed to the values given in the metadata file.

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 5]
```diff
@@ -22,27 +22,12 @@
...
-            <!--<class name="org.wso2.identity.integration.test.user.profile.mgt.NotificationOnUserOperationTestCase"/>-->
-            <!--<class name="org.wso2.identity.integration.test.entitlement.EntitlementNotificationTestCase"/>-->
-        <!--</classes>-->
+    <!--<classes>-->
+    <!--<class name="org.wso2.identity.integration.test.user.profile.mgt.NotificationOnUserOperationTestCase"/>-->
+    <!--<class name="org.wso2.identity.integration.test.entitlement.EntitlementNotificationTestCase"/>-->
+    <!--</classes>-->
     <!--</test>-->
 
-    <test name="is-tests-usr-mgt" preserve-order="true" parallel="false">
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Why is this removed
